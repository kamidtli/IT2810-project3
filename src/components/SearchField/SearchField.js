import React, { useState } from 'react';
import { connect } from 'react-redux';
import { TextField } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Redirect } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
    margin: 'auto',
    textAlign: 'center',
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  dense: {
    marginTop: theme.spacing(2),
  },
  menu: {
    width: 200,
  },
}));

function SearchField(props) {
  const classes = useStyles();
  const [toResults, setToResults] = useState(false);
  const [query, setQuery] = useState('');

  const handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      event.target.blur();
      setQuery(event.target.value);
      setToResults(true);
      props.resetSearch();
      props.addSearch(event.target.value);
      event.preventDefault();
    }
  };

  if (toResults) {
    return <Redirect push to={`/search?q=${query}`} />;
  }
  return (
    <form className={classes.container} noValidate autoComplete="off">
      <TextField
        id="outlined-with-placeholder"
        label="Search"
        className={classes.textField}
        margin="normal"
        onKeyDown={handleKeyDown}
        variant="outlined"
      />
    </form>
  );
}

// Empty because we don't need props here, but need the function in 'connect'
const mapStateToProps = (state) => ({

});

const mapDispatchToProps = (dispatch) => ({
  addSearch: (searchString) => dispatch({ type: 'NEW_SEARCH', searchString }),
  resetSearch: () => dispatch({ type: 'RESET_SEARCH' }),
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchField);
