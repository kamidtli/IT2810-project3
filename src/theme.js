import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#F6AE2D',
    },
    secondary: {
      main: '#000d34',
    },
  },
  status: {
    danger: 'orange',
  },
});

const responsiveTheme = responsiveFontSizes(theme);

export default responsiveTheme;
