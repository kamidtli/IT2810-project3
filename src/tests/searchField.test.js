import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from '@material-ui/styles';
import SearchField from '../components/SearchField/SearchField';
import theme from '../theme';

describe('Searchfield component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ThemeProvider theme={theme}><SearchField /></ThemeProvider>);
    expect(wrapper.toJSON).toMatchSnapshot();
  });
});
