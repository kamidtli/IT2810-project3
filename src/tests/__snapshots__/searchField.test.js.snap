// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchfield component renders without crashing 1`] = `undefined`;
